### PR TITLE
Ability to bootstrap a pre-release version of vcpkg-tool

### PR DESCRIPTION
**Pull request by michaelbprice**
This PR adds switch to the Unix bootstrap shell script that instructs it to build the vcpkg binary from the HEAD of the default branch for vcpkg-tool.

This allows users to opt-in to bootstrap a vcpkg binary before it is included in a release. This is primarily useful for early adopter testing that may happen between cuts of official releases, but it presents the possibility of further extension so that users can test their change to branches of vcpkg-tool in an easier manner.

### CODE REVIEW

**Comment by BillyONeal** [comment]
I don't necessarily oppose this but I would rather resolve it by just doing tool releases wherever necessary.

**Comment by BillyONeal** [comment]
Given that it's clear that we can just do releases, should we close this?

**Comment by michaelbprice** [comment]
TL;DR - Yeah, I'm fine with closing it.

The addition of the text file to control which released versions get bootstrapped solved one half of what this was looking to solve. The issue of being able to easily bootstrap a version of the tool that is off in someone else's PR branch still exists, which this would solve.

I don't see that as a great reason to keep this open though. I'm comfortable keeping these changes in a branch of my fork in case I want to make further improvements and either submit a new PR or reopen this one.

**Comment by BillyONeal** [comment]
Thanks!
